### DISCUSSION TITLE
How to deform with a mesh

### DISCUSSION THREAD

**Discussion by AeroDEmi** [python-pillow/Pillow#6452] [Q&A] [answered]
Hi guys,

I have a question, I have a mesh with deformations, like this:
![image](https://user-images.githubusercontent.com/44657733/179849226-a33a4132-b0a7-437e-b2da-433478d44222.png)

and I would like to deform an image with this mesh. I found that Image.Image.transform has (target , source) but the target is a square while the source is a quadrilateral shape. Is there a way to make the target quadrilateral and the source square?

**Comment by radarhere** [comment, marked as answer]
What leads you to the conclusion that https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.transform requires the target to be a square?

**Reply by radarhere** [reply]
The following code transforms our square test image into a quadrilateral target.

```python
from PIL import Image
im = Image.open("hopper.png")
w, h = im.size
out = im.transform(im.size, Image.Transform.QUAD, (-w//2, -h//2, 0, h, w*2, h*2, w, 0), Image.Resampling.BILINEAR)
```
![hopper](https://user-images.githubusercontent.com/3112309/182371213-c7a34d87-b4db-4d25-85a7-1f1fe0668858.png)
![transformed](https://user-images.githubusercontent.com/3112309/182371298-1e1aca5e-05f6-4f2a-90b6-1d65eb8571f1.png)
